### PR TITLE
[SPARK-39345][CORE][SQL][DSTREAM][ML][MESOS][SS] Replace `filter(!condition)` with `filterNot(condition)` 

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr replace `filter(!condition)` with `filterNot(condition)` .

### Why are the changes needed?
Use appropriate api.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass Github Actions

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala [hunk 1]
```diff
@@ -413,7 +413,7 @@ class SQLAppStatusListener(
       if (other != exec) other.stages else Nil
     }.toSet
     stageMetrics.keySet().asScala
-      .filter(!activeStages.contains(_))
+      .diff(activeStages)
```
Comment (LuciferYang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala:416]: `.filter(!activeStages.contains(_)) ` 
-> 
`.filterNot(activeStages.contains)`
->
`.diff(activeStages)`

**Comment by LuciferYang** [comment]
cc @wangyum

**Comment by wangyum** [comment]
cc @srowen

**Review by srowen — COMMENTED** [review]
The change is OK; I don't think it adds any performance. The only hesitation here would be code churn and possible merge conflicts due to this. I'm kind of neutral on it, I think it's OK.

**Comment by LuciferYang** [comment]
> The change is OK; I don't think it adds any performance. The only hesitation here would be code churn and possible merge conflicts due to this. I'm kind of neutral on it, I think it's OK.

If this acceptable, I can give pr for 3.3, 3.2 and 3.1

**Comment by srowen** [comment]
No, we wouldn't backport this, that's more change. 
Does this offer any benefit? I'm not sure it's more readable even.

**Pull request closed without merging**